p2p/switch.go

```diff
@@ -317,22 +335,24 @@ func (sw *Switch) DialSeeds(addrBook *AddrBook, seeds []string) error {
 		addrBook.Save()
 	}
 
-	// Ensure we have a completely undeterministic PRNG. cmd.RandInt64() draws
-	// from a seed that's initialized with OS entropy on process start.
-	rng := rand.New(rand.NewSource(cmn.RandInt64()))
-
 	// permute the list, dial them in random order.
-	perm := rng.Perm(len(netAddrs))
+	perm := sw.rng.Perm(len(netAddrs))
 	for i := 0; i < len(perm); i++ {
 		go func(i int) {
-			time.Sleep(time.Duration(rng.Int63n(3000)) * time.Millisecond)
+			sw.randomSleep(0)
 			j := perm[i]
 			sw.dialSeed(netAddrs[j])
 		}(i)
 	}
 	return nil
 }
 
+// sleep for interval plus some random amount of ms on [0, dialRandomizerIntervalMilliseconds]
+func (sw *Switch) randomSleep(interval time.Duration) {
+	r := time.Duration(sw.rng.Int63n(dialRandomizerIntervalMilliseconds)) * time.Millisecond
```

Comment:
Am not sure that this is exponential backoff:
It should be `(1 << n) * unitOfTime e.g millisecond` + `jitter from rng`
this one only has the `jitter from rng`

This code instead seems to sleep just a constant amount of times which defeats the purpose of exponentially backing off.

Reply:
this is just a function that sleeps an input+rand amount.

its up to the caller to ensure the passed value is exponentially backing off, ie https://github.com/tendermint/tendermint/pull/955/files#diff-44d98df4858f140bb01702ae7982ec31R476

Legit ?

Reply:
Oh I see, my bad, awesome work!